src/chart/ui/chart.h

```diff
@@ -56,8 +71,9 @@ class ChartWidget : public GUI::Widget
 	bool unprocessedPointerLeave{};
 	std::optional<int64_t> trackedMarkerId;
 	std::optional<int64_t> reportedMarkerId;
+	bool needUpdate{true};
```

Comment:
Signalling the need for an update is redundant in this class. this is what the doChange callback for.
needUpdate was part of the JS specific solution to assync connect the chart's onchange to the rendering call.

Reply:
We call `Interface::update` periodically
```
      this._updateInterval = setInterval(() => {
        this.render.updateFrame()
      }, 25)
```
So this should be refactored too. I think this is currently not the scope of the PR.

Reply:
I created a ticket from this discussion